### PR TITLE
refactor!: split `r/demo/boards`

### PR DESCRIPTION
**Pull request by harry-hov**
BREAKING CHANGE!

Found another `p` which imports `r`. It `should not happen`.

This PR splits `r/demo/boards` into `r/demo/boards` and `p/demo/boards`. Now `p/demo/groups` imports `p/demo/boards` instead of `r/demo/boards`

Related: #1393

### CODE REVIEW

File: examples/gno.land/r/demo/boards/board.gno [hunk 1]
```diff
@@ -1,140 +1,30 @@
...
-		}
-	}
-	return false
-}
-
 // Renders the board for display suitable as plaintext in
 // console.  This is suitable for demonstration or tests,
 // but not for prod.
-func (board *Board) RenderBoard() string {
+func RenderBoarddddddd(board *boards.Board) string {
```
Comment (ltzmaxwell) [inline examples/gno.land/r/demo/boards/board.gno:10]: typo?

Comment (harry-hov) [reply examples/gno.land/r/demo/boards/board.gno]: haha yeah :D

Fixed it!

File: examples/gno.land/p/demo/boards/board.gno [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+func (board *Board) GetThread(pid PostID) *Post {
+	pidkey := postIDKey(pid)
+	postI, exists := board.threads.Get(pidkey)
+	if !exists {
+		return nil
+	}
+	return postI.(*Post)
+}
+
+func (board *Board) AddThread(creator std.Address, title string, body string) *Post {
```
Comment (ltzmaxwell) [inline examples/gno.land/p/demo/boards/board.gno:94]: the name `thread` and `post` has been used interchangeably.

File: examples/gno.land/p/demo/boards/misc.gno [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package boards
+
+import (
+	"strconv"
+	"strings"
+)
+
+func padLeft(str string, length int) string {
```
Comment (ltzmaxwell) [inline examples/gno.land/p/demo/boards/misc.gno:8]: this is not used anywhere.

File: examples/gno.land/p/demo/boards/board.gno (hunk 2 above)
Comment (harry-hov) [reply examples/gno.land/p/demo/boards/board.gno:94]: `thread` is basically a `post` (feel free to correct me, I didn't get your concern)

I just copied it over without any modifications 🤷‍♂️

File: examples/gno.land/p/demo/boards/misc.gno (hunk 3 above)
Comment (harry-hov) [reply examples/gno.land/p/demo/boards/misc.gno]: Just copied this one as well. Seems like it wasn't used in the original implementation.  

Removed!

File: examples/gno.land/p/demo/boards/board.gno (hunk 2 above)
Comment (ltzmaxwell) [reply examples/gno.land/p/demo/boards/board.gno:94]: yeah mostly nits,  I think you can keep it as it were.

Comment (Kouteki) [reply examples/gno.land/p/demo/boards/board.gno:94]: @ltzmaxwell could you approve this PR, or request changes if needed?

**Pull request closed without merging**